components/progress-bar/script.js

```diff
@@ -149,9 +186,30 @@ class ProgressBar extends BaseComponent {
             }
 
             this.hasStarted = true;
-            this.startAnimation();
+            this.startAnimation(fromBeginning);
         } else {
-            this.filler.style.width = this.targetValue + '%';
+            this._fillerWidth = this.targetValue;
+        }
+    }
+
+    /**
+     * Custom element lifecycle method. Called when an attribute is changed.
+     * @param {string} name
+     * @param {string} oldValue
+     * @param {string|boolean} newValue
+     */
+    attributeChangedCallback(name, oldValue, newValue) {
+        if (!this.isRendered) return;
+        if (!components.isNumberPositiveValidation(name, newValue)) return;
+
+        if (name === 'target-value') {
+            this._targetValue = newValue;
+            this.setProgress();
+        }
+
+        if (name === 'animation-duration') {
+            this._animDuration = parseInt(newValue);
+            this.setProgress(true);
```

Comment:
Maybe add a return statement here so it will skip the second check if the first one is successful. Or use `switch` instead.